tests/prototypes/random_data_providers.py

```diff
@@ -0,0 +1,113 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2023 Scipp contributors (https://github.com/scipp)
+from __future__ import annotations
+
+from typing import Any, Generator, List, NewType
+
+import scipp as sc
+from numpy.random import Generator as RNG
+
+from beamlime.constructors import ProviderGroup
+
+from .parameters import EventRate, FrameRate, NumFrames, NumPixels, RandomSeed
+
+# Derived Configuration
+EventFrameRate = NewType("EventFrameRate", int)  # [events/frame]
+
+# Generated Data
+DetectorCounts = NewType("DetectorCounts", sc.Variable)
+TimeCoords = NewType("TimeCoords", dict[str, sc.Variable])
+RandomPixelId = NewType("RandomPixelId", sc.Variable)
+RandomEvent = NewType("RandomEvent", sc.DataArray)
+RandomEvents = NewType("RandomEvents", List[sc.DataArray])
+
+
+def provide_rng(random_seed: RandomSeed) -> RNG:
+    from numpy.random import default_rng
+
+    return default_rng(random_seed)
+
+
+def calculate_event_per_frame(
+    frame_rate: FrameRate, event_rate: EventRate
+) -> EventFrameRate:
+    return EventFrameRate(int(event_rate / frame_rate))
+
+
+def provide_time_coords(rng: RNG, ef_rate: EventFrameRate) -> TimeCoords:
+    dummy_zeros = [zr for zr in range(int(13620492e11), int(13620492e11) + ef_rate)]
```

Comment:
where does the number `13620492e11` come from?

Reply:
It's just a random datetime I picked... `2013-02-28T12:00:00` : D.... Actually event_time_zero was not used in this dummy workflow. Maybe it should be...